pupil-spa/src/app/services/check-complete/check-complete.service.ts

```diff
@@ -0,0 +1,111 @@
+import { Injectable } from '@angular/core';
+import { APP_CONFIG } from '../config/config.service';
+import {
+  CheckSubmissionApiCalled,
+  CheckSubmissionAPIFailed,
+  CheckSubmissionAPICallSucceeded,
+} from '../audit/auditEntry';
+import { AzureQueueService } from '../azure-queue/azure-queue.service';
+import { AuditService } from '../audit/audit.service';
+import { SubmissionService } from '../submission/submission.service';
+import { StorageService } from '../storage/storage.service';
+import { TokenService } from '../token/token.service';
+import { Router } from '@angular/router';
+
+/**
+ * Declaration of check start service
+ */
+@Injectable()
+export class CheckCompleteService {
+
+  featureUseHpa;
+  checkSubmissionApiErrorDelay;
+  checkSubmissionAPIErrorMaxAttempts;
+  submissionPendingViewMinDisplay;
+
+  constructor(private azureQueueService: AzureQueueService,
+              private submissionService: SubmissionService,
+              private storageService: StorageService,
+              private tokenService: TokenService,
+              private auditService: AuditService,
+              private router: Router) {
+    const { featureUseHpa,
+      checkSubmissionApiErrorDelay,
+      checkSubmissionAPIErrorMaxAttempts,
+      submissionPendingViewMinDisplay,
+    } = APP_CONFIG;
+    this.featureUseHpa = featureUseHpa;
+    this.checkSubmissionApiErrorDelay = checkSubmissionApiErrorDelay;
+    this.checkSubmissionAPIErrorMaxAttempts = checkSubmissionAPIErrorMaxAttempts;
+    this.submissionPendingViewMinDisplay = submissionPendingViewMinDisplay;
+  }
+
+  private sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  /**
+   * Check complete submission
+   * @param {Number} startTime
```

Comment:
Please describe what `startTime` is used for and what units it should be in.

Reply:
@jon-shipley included.